br/pkg/lightning/mydump/region.go

```diff
@@ -310,13 +311,18 @@ func makeParquetFileRegion(
 	dataFile FileInfo,
 	prevRowIdxMax int64,
 ) (int64, *TableRegion, error) {
-	r, err := store.Open(ctx, dataFile.FileMeta.Path)
-	if err != nil {
-		return prevRowIdxMax, nil, errors.Trace(err)
-	}
-	numberRows, err := ReadParquetFileRowCount(ctx, store, r, dataFile.FileMeta.Path)
-	if err != nil {
-		return 0, nil, errors.Trace(err)
+	var numberRows int64 = 0
+
+	// for S3, reading an empty file will cause error, so skip reading the empty files
+	if dataFile.FileMeta.FileSize > 0 {
```

Comment:
I still think we should skip directory like object in cloud storage instead of writing this special logic for specific type of files. It's import to keep each external storage's behavior the same as much as we can.

Reply:
I did a test that for parquet file importing, not only an empty directory but also an empty file in S3 will cause the error.  So any empty objects will trigger the problem, and we cannot just stick on handling empty directories.

Reply:
> I did a test that for parquet file importing, not only an empty directory but also an empty file in S3 will cause the error. So any empty objects will trigger the problem, and we cannot just stick on handling empty directories.

I think this is expected behavior. We should only exclude empty directory, but for empty file, it's the user's responsibility to exclude them by writing the right regexp if needed.